app/src/components/form/attribute_input.rs

```diff
@@ -68,3 +69,59 @@ pub fn single_attribute_input(props: &SingleAttributeInputProps) -> Html {
         </div>
     }
 }
+
+#[derive(Properties, PartialEq)]
+pub struct ListAttributeInputProps {
+    pub name: String,
+    pub attribute_type: AttributeType,
+    pub values: Vec<String>,
+}
+
+#[function_component(ListAttributeInput)]
+pub fn list_attribute_input(props: &ListAttributeInputProps) -> Html {
+
+    // let value_indices = use_memo(props.values, |vals| use_state(|| (0..(vals.len() as isize)).collect::<Vec<_>>()));
+    let value_indices = use_state(|| (0..(props.values.len() as isize)).collect::<Vec<_>>() ); // use_memo(props.values)
+    let new_index = use_mut_ref::<isize>(|| 0);
+
+    html! {
+        <div class="row mb-3">
+            <label for={props.name.clone()}
+                class="form-label col-4 col-form-label"
+                title={props.name.clone()}>
+                {props.name[0..1].to_uppercase() + &props.name[1..].replace("_", " ")}{":"}
+            </label>
+            <div class="col-8">
+            {value_indices.iter().map(|i| {let i = *i; html! {
+                <div class="input-group mb-2" key={i}>
+                <AttributeInput
+                    attribute_type={props.attribute_type.clone()}
+                    name={props.name.clone()}
+                    value={usize::try_from(i).ok().and_then(|i| props.values.get(i)).cloned().unwrap_or_else(|| String::from(""))} />
```

Comment:
`unwrap_or_default`

Reply:
Also, it's reasonable to `.expect()` here, since the `i` are coming from `usize` to start with.

Reply:
So, just `i as usize` should work?

Reply:
`i` can be negative, and I don't wish to imagine the .expect() errors in case someone changes the underlying props.values list. Until we have it with a memo, or_default is better, and even after that we need to change for positive values.
...as I said, I'm not too happy with how value_indices turned out :sweat_smile:

Reply:
Turns out, I'm a bit silly and didn't realize I can just use indices past the array length for the same result.